lib/rules/no-unstable-nested-components.js

```diff
@@ -35,7 +35,7 @@ function generateErrorMessageWithParentName(parentName) {
  * @returns {Boolean}
  */
 function startsWithRender(text) {
-  return (text || '').startsWith('render');
+  return text !== undefined && text.startsWith('render');
```

Comment:
I'm specifically interested in changes related to this rule. Thanks for testing the performance @willheslam!

While this improvement is definitely welcome I'm having a hard time believing it actually speeds up this rule by ~40%.
Is it really that slow to create new Strings? This method is not even accessed that often. 
Why shouldn't other Strings also be declared once in module scope, e.g. `render`, `CallExpression`?

> Rule                                       | Time (ms) | Relative
> :------------------------------------------|----------:|--------:
> react/no-unstable-nested-components        |   172.198 |     2.7%
> 
> to
> 
> Rule                                       | Time (ms) | Relative
> :------------------------------------------|----------:|--------:
> react/no-unstable-nested-components        |    95.287 |     2.3%

Reply:
Right, that makes perfect sense. Those utilities are perfect spot for optimization as they are being used widely across the code base. 
That's impressive improvement! 💯